libafl_bolts/src/shmem.rs

```diff
@@ -204,52 +207,53 @@ pub trait ShMem: Sized + Debug + Clone + AsSlice<Entry = u8> + AsMutSlice<Entry
         self.len() == 0
     }
 
-    /// Convert to an owned object reference
+    /// Convert to a [`OwnedRef`] of a given type, checking that the size is correct
     ///
     /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the object with something like
-    /// `*shmem.as_object_mut::<T>() = T::new();`
-    unsafe fn as_object<T: Sized + 'static>(&self) -> &T {
+    /// This function is not safe if the type was not previously initialized
+    /// or a different type may be placed at this memory location.
+    /// Further users of this [`OwnedRefMut`] will access the underlying data directly.
+    unsafe fn as_owned_ref_of<'a, T: Sized + 'static>(&self) -> OwnedRef<'a, T> {
         assert!(self.len() >= core::mem::size_of::<T>());
-        (self.as_slice().as_ptr() as *const () as *const T)
-            .as_ref()
-            .unwrap()
+        OwnedRef::from_ptr(self.as_slice().as_ptr() as *const T)
     }
 
-    /// Convert to an owned object mutable reference
+    /// Convert to a [`OwnedRefMut`] of a given type, checking that the size is correct
     ///
     /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the object with something like
-    /// `*shmem.as_object_mut::<T>() = T::new();`
-    unsafe fn as_object_mut<T: Sized + 'static>(&mut self) -> &mut T {
+    /// This function is not safe if the type was not previously initialized
+    /// or a different type may be placed at this memory location.
+    /// Further users of this [`OwnedRefMut`] will access the underlying data directly.
+    unsafe fn as_owned_ref_mut_of<'a, T: Sized + 'static>(&mut self) -> OwnedRefMut<'a, T> {
         assert!(self.len() >= core::mem::size_of::<T>());
-        (self.as_mut_slice().as_mut_ptr() as *mut () as *mut T)
-            .as_mut()
-            .unwrap()
+        OwnedRefMut::from_mut_ptr(
+            (self.as_mut_slice().as_mut_ptr() as *mut () as *mut T)
+                .as_mut()
+                .unwrap(),
```

Comment:
You've already got a `*mut T` at this point:
```suggestion
            self.as_mut_slice().as_mut_ptr() as *mut () as *mut T
```

Reply:
Was an oversight, thanks! But anyway, changed the API changed again :)